monai/losses/perceptual.py

```
@@ -84,13 +95,18 @@ def __init__(
-            raise ValueError(
-                "MedicalNet networks are only compatible with ``spatial_dims=3``."
-                "Argument is_fake_3d must be set to False."
-            )
 
-        if channel_wise and "medicalnet_" not in network_type:
+        # Strict validation for MedicalNet
+        if "medicalnet_" in network_type:
+            if spatial_dims == 2 or is_fake_3d:
+                raise ValueError(
+                    "MedicalNet networks are only compatible with ``spatial_dims=3``. Argument is_fake_3d must be set to False."
+                )
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)